Clojure/Clojurescript support via Planck (~1 sec lag) (fixes #34)

cirqueit opened the pull request:
Using [planck](http://planck-repl.org/). Originally tried several alternatives

- `lein repl` takes much longer
- `grench repl` was instant but requires a `project.clj` file, starting up a headless REPL to (re)connect to (state issues) and can't parse multiline s-expressions.

In the spirit of k.i.s.s, I think this is the way to go, though launching [planck's socket REPL](http://planck-repl.org/socket-repl.html) and (re)connecting to it may make things faster (currently not in their Linux release [yet](https://github.com/mfikes/planck/pull/297), and I don't has OSX to test that out).

metakirby5 commented:
Excellent! Thank you so much for the contribution :)

Would you mind also adding Clojure info to the relevant sections of the README and documentation?

metakirby5 commented on autoload/codi/load.vim:
```diff
@@ -62,6 +62,10 @@ let s:codi_default_interpreters = {
+      \ 'clojure': {
+          \ 'bin': ['planck', '--verbose', '--dumb-terminal'],
+          \ 'prompt': '^.\{-}=> ',
           \ }
```
I didn't notice it before, but this should really have a comma to be consistent with the rest of the dicts. I can change it myself if you like, but since you're already touching this it may just be easier to change it in this PR.

cirqueit commented:
I've added the trailing comma, and updated the `README` and `doc.txt`.
Hopefully the socket REPL will speed things up in the future, but for now at least Clojure is supported.
